### ISSUE TITLE
Shell integration: Use command tracking as alternative child process tracking

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#139522]
Use command tracking to save running a process for whether the shell is dirty, similar to cwd tracking this will lead to more responsive UI and less work happening on update. https://github.com/microsoft/vscode/blob/a1bf964306d30a9bbccb404b1b653e5a02fe3052/src/vs/platform/terminal/node/childProcessMonitor.ts

**Comment by meganrogge** [comment]
Blocked by supporting subshells

**Comment by Tyriar** [comment]
Created the subshell issue https://github.com/microsoft/vscode/issues/141343, instead of relying on sub-shell support, we could just say when the shell is not in the prompt, the terminal's busy. This is arguably better than the current child process support anyway as the terminal is doing something. Sub-shells are a problem as it would always be busy, but that's a problem that could be handled with #131343.

Need to way the pros/cons for both ways.

**Comment by Tyriar** [comment]
This is mainly a small performance improvement where we would be able to not poll for the process tree as much.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!
